### PR TITLE
feat: Rework readme file

### PR DESCRIPTION
**Pull request by Hafflgav**
This commit includes a rework of the readme file 
Now it should be... 
* structured better
* more expressive 
... which makes contributing easier.

### CODE REVIEW

**Comment by Hafflgav** [comment]
Would be cool to get this merged soon since I am awaiting a repo review by algolia to get search functionality for our open source project.

File: README.md [hunk 1]
```diff
@@ -1,17 +1,35 @@
-# Miranum-Website
+# Miranum Documentation 
 
-This repository contains the code our website is build with.
-The life version can be found under [https://miranum.com/](https://miranum.com/)
+This repository contains the Miranum documentation with can be found under [https://miranum.com/](https://miranum.com/)
```
Comment (rami-nk) [inline README.md:3]: which instead of with

File: README.md [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
-You'd like to contribute? You're welcome!
+### Structure
+Name Markdown files according to the title. This makes it easier to find a file.
+Avoid non-alphanumeric characters in titles. Use the file name as internal document id to reference in the appropriate sidebars file.
 
-This website is built using [Docusaurus 2](https://docusaurus.io/), a modern static website generator.
+### Style Guide 
+We will be using the writing [style guide defined by Camunda](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/technical-writing-styleguide.md).
+It outlines writing techniques and practices to ensure uniform styling across documentation and to yield a more cohesive and organized user experience.
+We also want to ensure a unified naming of our product components. Hence, please use **Miranum Connect**, **Miranum IDE**, **Miranum: Modeler**, **Miranum: Form Builder** and **Miranum: Console** if you are referring to them.
```
Comment (lmoesle) [inline README.md:18]: Should we name the VS-Code Extension with `:`?

```suggestion
We also want to ensure a unified naming of our product components. Hence, please use **Miranum Connect**, **Miranum IDE**, **Miranum Modeler**, **Miranum Form Builder** and **Miranum Console** if you are referring to them. 
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,35 @@
...
 
-This repository contains the code our website is build with.
-The life version can be found under [https://miranum.com/](https://miranum.com/)
+This repository contains the Miranum documentation with can be found under [https://miranum.com/](https://miranum.com/)
 
-## development
+## Documentation Guidelines
+### PRs for every change
+All changes have to be done in a separate Branch. As soon as the changes are done please open a PR. A GitHub Action runs 
+with every commit to a Branch and checks if the documentation can be build. If you create a new branch make sure to name him according
```
Comment (rami-nk) [inline README.md:8]: * can be built
* name "it" according to what "it does" / or "you do"

File: README.md [hunk 4]
```diff
@@ -1,17 +1,35 @@
...
-This repository contains the code our website is build with.
-The life version can be found under [https://miranum.com/](https://miranum.com/)
+This repository contains the Miranum documentation with can be found under [https://miranum.com/](https://miranum.com/)
 
-## development
+## Documentation Guidelines
+### PRs for every change
+All changes have to be done in a separate Branch. As soon as the changes are done please open a PR. A GitHub Action runs 
+with every commit to a Branch and checks if the documentation can be build. If you create a new branch make sure to name him according 
+to what you to (e.g. feat/xyz or fix/xyz). Please use semantic commit messages as described [here](https://gist.github.com/joshbuchea/6f47e86d2510bce28f8e7f42ae84c716).
```
Comment (lmoesle) [inline README.md:9]: I would add that to a contributing.md. You may also add the naming conventions to the pull_request_template.md

**Review by lmoesle — APPROVED** [review]
Looks good to me

File: README.md (hunk 4 above)
Comment (rami-nk) [inline README.md:9]: As described "in" / or as described "here"

File: README.md [hunk 5]
```diff
@@ -1,17 +1,35 @@
...
 
-You'd like to contribute? You're welcome!
+### Structure
+Name Markdown files according to the title. This makes it easier to find a file.
+Avoid non-alphanumeric characters in titles. Use the file name as internal document id to reference in the appropriate sidebars file.
```
Comment (rami-nk) [inline README.md:13]: Use the filename as "an" internal document id to reference it in the appropriate sidebar file.

File: README.md (hunk 2 above)
Comment (Hafflgav) [reply README.md:18]: Good point @lmoesle - I have thought about this and we might want to talk about it. Imho it is important to have consistent naming. If we are not using the ":" we should remove it from our VS Code Plugins as well which I do not think we want to... 
(But I do agree it does not look to nice in the documentation)

File: README.md [hunk 6]
```diff
@@ -1,17 +1,35 @@
...
 
-## deployment
+### Local Development 
+`npm run start`
 
-Our public website is deployed on netlify. Create a pull request and after our review you'll see your changes online.
\ No newline at end of file
+### Troubleshooting checklist
+Have you pulled latest from `main`?
+Have you run npm install? When we update dependencies in the project, they don't automatically update in your environment.
```
Comment (rami-nk) [inline README.md:29]: they don't automatically get updated in your environment

**Review by rami-nk — COMMENTED** [review]
Here are some typos I found. Please fix before merging

File: README.md (hunk 4 above)
Comment (Hafflgav) [reply README.md:9]: I took a peak into some of our other projects and non of them has a separate contributing.md. Hence I would leave it out for this repository as well. 
The pull_request_template.md sounds like an excellent idea to add 👍

File: README.md (hunk 2 above)
Comment (lmoesle) [reply README.md:18]: Well we are still at an early stage. Renaming the extensions may be unpleasant but it's better to do it now than later. If you have a good suggestion I could rename the extension we currently have and add the naming conventions to our tech docs.

Comment (Hafflgav) [reply README.md:18]: I would say that we should not use any special characters like ":" in the product names. I am fine with Miranum Modeler and co. 👍

File: README.md (hunk 4 above)
Comment (Hafflgav) [reply README.md]: I just added the `pull_request_template.md` - would you mind checking it out real quick?

**Review by rami-nk — APPROVED** [review]

**Review by DaAnda97 — APPROVED** [review]

**Pull request merged**